kymatio/scattering1d/backend/numpy_backend.py

```diff
@@ -49,46 +49,15 @@ def pad_1d(x, pad_left, pad_right, mode='constant', value=0.):
     pad_right : int
         amount to add on the right of the tensor (at the end of the temporal
         axis).
-    mode : string, optional
-        Padding mode. Options include 'constant' and 'reflect'. See the
-        PyTorch API for other options.  Defaults to 'constant'.
-    value : float, optional
-        If mode == 'constant', value to input within the padding. Defaults to
-        0.
     Returns
     -------
-    res : tensor
+    output : tensor
         The tensor passed along the third dimension.
     """
     if (pad_left >= x.shape[-1]) or (pad_right >= x.shape[-1]):
-        if mode == 'reflect':
-            raise ValueError('Indefinite padding size (larger than tensor).')
-
-    res = np.pad(x, ((0, 0), (0, 0), (pad_left, pad_right),), mode=mode)
-    return res
-
+        raise ValueError('Indefinite padding size (larger than tensor).')
 
-def pad(x, pad_left=0, pad_right=0):
-    """Pad real 1D tensors and map to complex
-    Padding which allows to simultaneously pad in a reflection fashion and map
-    to complex if necessary.
-    Parameters
-    ----------
-    x : tensor
-        Three-dimensional input tensor with the third axis being the one to
-        be padded.
-    pad_left : int
-        Amount to add on the left of the tensor (at the beginning of the
-        temporal axis).
-    pad_right : int
-        amount to add on the right of the tensor (at the end of the temporal
-        axis).
-    Returns
-    -------
-    output : tensor
-        A padded signal.
-    """
-    output = pad_1d(x, pad_left, pad_right, mode='reflect')
+    output = np.pad(x, ((0, 0), (0, 0), (pad_left, pad_right),), mode='reflect')
```

Comment:
Can you make this as general as the padding function in the tf backend? (but using tuples instead of lists?)

Reply:
cf https://github.com/kymatio/kymatio/pull/573/files#diff-e3f2ae7d9ddd5b7b89028e91fe70b247R60

Reply:
I think I can do it for the Numpy. I tried for the pytorch, but it was getting a bit complicated/wasn't working.